src/main/java/seedu/duke/assets/MedicineList.java

```diff
@@ -95,4 +98,59 @@ public void edit(String[] parameterArray) throws NotFoundException {
     public ArrayList<Medicine> getList() {
         return medicines;
     }
+
+    public void findById(String[] parameters) {
+        this.returnedFinderArray = MedicineFinder.findMedicineById(medicines, parameters[1]);
+    }
+
+    public void findByDosage(String[] parameters) {
+        this.returnedFinderArray = MedicineFinder.findMedicineByDosage(medicines, Integer.parseInt(parameters[1]));
+    }
+
+    public void findByExpiry(String[] parameters) {
+        this.returnedFinderArray = MedicineFinder.findMedicineByExpiry(medicines, parameters[1]);
+    }
+
+    public void findBySideEffects(String[] parameters) {
+        this.returnedFinderArray = MedicineFinder.findMedicineBySideEffects(medicines, parameters[1]);
+    }
+
+    public void findByQuantity(String[] parameters) {
+        this.returnedFinderArray = MedicineFinder.findMedicineByQuantity(medicines, Integer.parseInt(parameters[1]));
+    }
+
+    public void findByName(String[] parameters) {
+        this.returnedFinderArray = MedicineFinder.findMedicineByName(medicines, parameters[1]);
+    }
+
+    public void findByNric(String[] parameters) {
+        // Intentionally left blank
+    }
+
+    public void findByAge(String[] parameters) {
+        // Intentionally left blank
+    }
+
+    public void findByGender(String[] parameters) {
+        // Intentionally left blank
+    }
+
+    public void findByAddress(String[] parameters) {
+        // Intentionally left blank
+    }
+
+    public void findByDob(String[] parameters) {
+        // Intentionally left blank
+    }
+
+    public void findBySpecialization(String[] parameters) {
+        // Intentionally left blank
+    }
+
+    public void findByDateAdmission(String[] parameters) {
+        // Intentionally left blank
+    }
```

Comment:
All these are irrelevant to Medicine?

Reply:
It's not but I need to put it there cos y'all implemented abstract in List so those that extended List.
Those "left blank" method essentially doesn't return anything. But can't remove it cos it's abstract subclass implementation